backstory/src/main/java/com/google/sps/servlets/StoryManagerImpl.java

```diff
@@ -0,0 +1,163 @@
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auth.oauth2.IdTokenCredentials;
+import com.google.auth.oauth2.IdTokenProvider;
+import java.io.IOException;
+import java.util.Scanner;
+import org.json.JSONObject;
+
+/**
+ * Generates text through interface with Cloud Contained
+ * GPT-2 Model.
+ */
+public final class StoryManagerImpl implements StoryManager {
+  /**
+   * Prefix, Maximum Length, and Temperature(Volatility) fields
+   */
+  private String prefix;
+  private int maxTextLength;
+  private Double temperature;
+
+  private StoryManagerRequestFactory requestFactory;
+
+  /**
+   * Instantiate StoryManager. Use "", 100-1000, and 1 as parameter
+   * defaults in case of invalid input
+   *
+   * @param prefix String to serve as generation prompt.
+   * @param maxLength Maximum text character length for generation output.
+   * @param temperature Double to hold number 0-1 for text generation volatility.
+   */
+  public StoryManagerImpl(String prefix, int maxLength, Double temperature) {
+    this.prefix = prefix;
+    this.maxTextLength = maxLength;
+    this.temperature = temperature;
+    requestFactory = new StoryManagerRequestFactoryImpl();
+
+    if (prefix == null) {
+      this.prefix = "";
+    }
+    if (maxLength < 100) {
+      this.maxTextLength = 100;
+    }
+    if (maxLength > 1000) {
+      this.maxTextLength = 1000;
+    }
+    if (temperature < 0 || temperature > 1) {
+      this.temperature = 1.0;
+    }
+  }
+
+  /**
+   * Returns generation prefix.
+   *
+   * @return String The generation prefix.
+   */
+  public String getPrefix() {
+    return prefix;
+  }
+
+  /**
+   * Returns maximum length for generation.
+   *
+   * @return int The maximum length for text generation.
+   */
+  public int getMaxLength() {
+    return maxTextLength;
+  }
+
+  /**
+   * Returns temperature(volatility of generation).
+   *
+   * @return Double Numerical quantity representing temperature.
+   */
+  public Double getTemperature() {
+    return temperature;
+  }
+
+  /**
+   * Makes a post request with a JSON including GPT2 Parameters
+   *
+   * @returns HttpResponse The reponse from the Generation server.
+   */
+  private HttpResponse makePostRequestGPT2() throws IOException {
+    // Form JSON body using generation parameters
+    String requestBody = makeRequestBody(prefix, maxTextLength, temperature);
+
+    // Build Request with Adapter and JSON Input
+    HttpRequest request = requestFactory.buildPostRequest(requestBody);
+    request.getHeaders().setContentType("application/json");
+
+    // Wait until response received
+    request.setConnectTimeout(0);
+    request.setReadTimeout(0);
+    return request.execute();
+  }
+
+  /**
+   * Returns generated text output using given fields.
+   *
+   * @return String Generated output text.
+   */
+  public String generateText() {
+    // Obtain response from Server POST Request
+    try {
+      HttpResponse outputResponse = makePostRequestGPT2();
```

Comment:
Do you get a different response each time `makePostRequestGPT2` is called? If not, maybe the response can be cached, so that successive calls to `generateTest()` can used the cached value rather than repeating the network call.

Reply:
The response will be random every time due to the nature of text generation.